packages/auth-sveltekit/src/server.ts

```diff
@@ -106,6 +106,34 @@ export default function serverAuth(client: Client, options: AuthOptions) {
   };
 }
 
+function setCookie(
+  cookies: Cookies,
+  value: string,
+  config: AuthConfig,
+  isAuth = false
```

Comment:
I think I prefer two separate functions here than a boolean with a default value. If you want to share the base configuration object between both functions, that's fine! Something like:

```ts
const BASE_COOKIE_CONFIG = {
  httpOnly: true,
  sameSite: "lax",
  path: "/",
};
```

then in the functions:

```ts
function setVerifierCookie(cookies: Cookies, config: AuthConfig, value: string): void {
  const expires = new Date(Date.now() + 1000 * 60 * 24 * 7); // in 7 days
  cookies.set(config.pkceVerifierCookieName, value, {
    ...BASE_COOKIE_CONFIG,
    expires,
    secure: config.baseUrl.startsWith("https"),
  });
}
```

Then in the class you can:

```ts
private setVerifierCookie(verifier: string) {
  setVerifierCookie(this.cookies, this.config, verifier);
}
```

---

The advantage here is that you can look at the call site and know which cookie it's setting without knowing if the optional `true` means verifier or auth token, etc.

Reply:
yes, I agree with all of these, will update : )